internal/manifests/collector/service.go

```diff
@@ -133,9 +135,10 @@ func Service(params manifests.Params) *corev1.Service {
 	}
 
 	// if we have no ports, we don't need a service
-	if len(ports) == 0 {
+	if len(ports) == 0 || err != nil {
```

Comment:
we've already done the error check at this point, so we can just revert this line